src/api/parser/src/parser-queue.js

```diff
@@ -0,0 +1,105 @@
+const { logger, fetch } = require('@senecacdot/satellite');
+const feedQueue = require('./feed/queue');
+const feedWorker = require('./feed/worker');
+const Feed = require('./data/feed');
+
+/**
+ * Adds the feed to the database if necessary, or gets a more complete
+ * version of the feed if we have better data already.
+ * @param {Object} feedData - feed data parsed from the wiki feed list.
+ * Returns Promise<Feed>, with the most appropriate Feed Object to use.
+ */
+async function updateFeed(feedData) {
+  let currentFeed;
+
+  // If we have an existing feed in the database for this URL, prefer that,
+  // since it might have updated cache info (e.g., etag).
+  const existingFeed = await Feed.byUrl(feedData.url);
+  if (existingFeed) {
+    // We have a version of this feed in the database already, prefer that
+    currentFeed = existingFeed;
+  } else {
+    // First time we're seeing this feed, add it to the database
+    const id = await Feed.create(feedData);
+    currentFeed = await Feed.byId(id);
+  }
+
+  return currentFeed;
+}
+
+/**
+ * Invalidates a feed
+ * @param feedData - Object containing feed data
+ */
+async function invalidateFeed(id) {
+  const feed = await Feed.byId(id);
+  // TODO: we need to bubble up the reason for the failure
+  await feed.setInvalid('unknown reason');
+  logger.info(`Invalidating feed ${feed.url} for the following reason: ${'unknown reason'}`);
+}
+
+/**
+ * Process all of these Feed objects into Redis and the feed queue.
+ * @param {Array<Feed>} feeds - the parsed feed Objects to be processed.
+ */
+function processFeeds(feeds) {
+  return Promise.all(
+    feeds.map(async (feed) => {
+      // Save this feed into the database if necessary.
+      const currentFeed = await updateFeed(feed);
+      // Add a job to the feed queue to process all of this feed's posts.
+      await feedQueue.addFeed({ id: currentFeed.id });
+    })
+  );
+}
+
+/**
+ * Download and parse feed author/URL info from the wiki and redis, and process
+ * these into Feed Objects to be added to the database and feed queue.
+ */
+async function processAllFeeds() {
+  try {
+    // Get an Array of Feed objects from the Users microservice and Redis
+    const res = await fetch(`${process.env.USER_URL}/`);
```

Comment:
For this to work we have to do a few things:

- In the docker-compose files, we have to declare that `USERS_URL` (note the trailing `S` on `USERS`) is part of the `environment` being passed into the process
- We need to specify the internal docker container domain for the localhost development case
- We need to create and pass a Service JWT Token to the Users Service so you can do this without being an authenticated user. 

You can checkout https://github.com/Seneca-CDOT/telescope/pull/2058/files, where I do similar things.

Reply:
Another thing we're going to need to do here is integrate a static JSON file with the current wiki feed list, so we can supplement the data (cc @manekenpix, @chrispinkney) with legacy feeds.  We probably need to do this in a follow-up PR vs. blocking this one on that, but FYI.